test/factory-test.js

```diff
@@ -183,7 +188,7 @@ describe('factory', function() {
           factory.build('person', { age: 30 }, function(err, person) {
             (person instanceof Person).should.be.true;
             person.should.not.have.property('saveCalled');
-            person.name.should.eql('Person 1');
```

Comment:
This probably should not be handled this way. A quick look at the tests suggests that the expectation is to reset nameCounter and emailCounter before each test. That doesn't happens because the counters are decraled inside the function.

You may move the declarations to the main describe method:
```

describe('factory', function() {
  var Model, Person, Job, Post;
  var nameCounter = 1, emailCounter = 1;
```

and then
```
  beforeEach(function() {
    nameCounter = 1;
    emailCounter = 1;
```

instead of

```
  beforeEach(function() {
    var nameCounter = 1, emailCounter = 1;
```